### PR TITLE
Add more network tests

### PR DESCRIPTION
**Pull request by dataclouder**
Add tests that use combinations of static IP and DHCP parameters where appropriate.
Extends issues #96, #97.

This PR could conflict with others that are adding documentation (the file `vcd.erb` is used by many).

I am not sure about the hyperlinks within `network.html.markdown`.

### CODE REVIEW

File: website/docs/r/network.html.markdown [hunk 1]
```diff
@@ -6,10 +6,12 @@ description: |-
   Provides a vCloud Director VDC Network. This can be used to create, modify, and delete internal networks for vApps to connect.
```
Comment (lvirbalas) [inline website/docs/r/network.html.markdown]: Please change all occurrences of "VDC Network" to "Org VDC Network", so it's clear that the context is organisation VDC as opposed to provider VDC.

File: website/docs/r/network_direct.html.markdown [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) A unique name for the network
+* `external_network` - (Required) The name of the external network.
+* `shared` - (Optional) Defines if this network is shared between multiple vDCs
+  in the vOrg.  Defaults to `false`.
+
```
Comment (lvirbalas) [inline website/docs/r/network_direct.html.markdown]: Missing `org` and `vdc` descriptions in all three doc files:

```
* `org` - (Optional; *v2.0+*) The name of organization to use, optional if defined at provider level. Useful when connected as sysadmin working across different organisations
* `vdc` - (Optional; *v2.0+*) The name of VDC to use, optional if defined at provider level
```

File: website/docs/r/network_direct.html.markdown [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Provides a vCloud Director VDC Network directly connected to an external network. This can be used to create,
+modify, and delete internal networks for vApps to connect.
+
+## Example Usage
+
+```hcl
+resource "vcd_network_direct" "net" {
+  org              = "my-org"
+  vdc              = "my-vdc"
```
Comment (lvirbalas) [inline website/docs/r/network_direct.html.markdown]: More people will use the adapter with org admin, as opposed to sys admin. So, let's mark `org` and `vdc` as optional and separate them by a new line (in all three examples). For example like this:

```
resource "vcd_network_direct" "net" {
  org              = "my-org" # Optional
  vdc              = "my-vdc" # Optional

  name             = "my-net"
  external_network = "my-ext-net"
}
```

File: website/docs/r/network.html.markdown (hunk 1 above)
Comment (dataclouder) [reply website/docs/r/network.html.markdown]: done

File: website/docs/r/network_direct.html.markdown (hunk 2 above)
Comment (dataclouder) [reply website/docs/r/network_direct.html.markdown]: done

File: website/docs/r/network_direct.html.markdown (hunk 3 above)
Comment (dataclouder) [reply website/docs/r/network_direct.html.markdown]: done

File: website/docs/r/network_direct.html.markdown [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+# vcd\_network\_direct (*v2.0+*)
+
+Provides a vCloud Director Org VDC Network directly connected to an external network. This can be used to create,
+modify, and delete internal networks for vApps to connect.
+
+## Example Usage
+
+```hcl
+resource "vcd_network_direct" "net" {
+  org              = "my-org"  #Optional
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply website/docs/r/network_direct.html.markdown]: fixed

**Pull request merged**